lib/app.rb

```diff
@@ -1,35 +1,68 @@
+require 'finder'
+require 'creator'
+
 class App
   attr_reader :contacts
 
-  def initialize(ui)
+  def initialize(ui, kernel = Kernel)
     @ui = ui
     @contacts = []
+    @kernel = kernel
   end
 
   FIELDS = [:first_name, :last_name, :email_address, :mobile_number, :twitter_handle]
```

Comment:
Yes, I guess the Creator would break first. Although I currently can't imagine what the wrong data would be. I'll ponder this a bit more...